pkg/api/common.go

```diff
@@ -133,10 +138,19 @@ func ParseFlags(fs *flag.FlagSet, cfg *Config) *Config {
 }
 
 func Validate(cfg *Config) error {
-	if cfg.EnableFeatures != "" {
-		cfg.EnabledFeaturesList = strings.Split(cfg.EnableFeatures, ",")
+	if cfg.PromscaleEnableFeatures != "" {
+		featureList := strings.Split(cfg.PromscaleEnableFeatures, ",")
+		cfg.PromscaleEnabledFeaturesMap = make(map[string]struct{}, len(featureList))
+		for _, f := range featureList {
+			cfg.PromscaleEnabledFeaturesMap[f] = struct{}{}
+		}
+	} else {
+		cfg.PromscaleEnabledFeaturesMap = make(map[string]struct{})
```

Comment:
Can we not do a `nil` since fetching a value for a key in a `nil` map also responds with false, so it should serve the same purpose.

Reply:
I have reworked this part of the code making this comment obsolete. I guess it would still be possible to work a `nil` map in there, but I'm not sure that it's worthwhile.